kv/task.go

```diff
@@ -278,78 +278,68 @@ func (s *Service) findTasks(ctx context.Context, tx Tx, filter influxdb.TaskFilt
 
 // findTasksByUser is a subset of the find tasks function. Used for cleanliness
 func (s *Service) findTasksByUser(ctx context.Context, tx Tx, filter influxdb.TaskFilter) ([]*influxdb.Task, int, error) {
-	if filter.User == nil {
-		return nil, 0, influxdb.ErrTaskNotFound
+	var ts []*influxdb.Task
+
+	taskBucket, err := tx.Bucket(taskBucket)
+	if err != nil {
+		return nil, 0, influxdb.ErrUnexpectedTaskBucketErr(err)
 	}
-	var org *influxdb.Organization
-	var err error
-	if filter.OrganizationID != nil {
-		org, err = s.findOrganizationByID(ctx, tx, *filter.OrganizationID)
-		if err != nil {
-			return nil, 0, err
-		}
-	} else if filter.Organization != "" {
-		org, err = s.findOrganizationByName(ctx, tx, filter.Organization)
+
+	var (
+		seek []byte
+		opts []CursorOption
+	)
+
+	if filter.After != nil {
+		seek, err = taskKey(*filter.After)
 		if err != nil {
 			return nil, 0, err
 		}
-	}
 
-	var ts []*influxdb.Task
+		opts = append(opts, WithCursorSkipFirstItem())
+	}
 
-	maps, err := s.findUserResourceMappings(
-		ctx,
-		tx,
-		influxdb.UserResourceMappingFilter{
-			ResourceType: influxdb.TasksResourceType,
-			UserID:       *filter.User,
-			UserType:     influxdb.Owner,
-		},
-	)
+	c, err := taskBucket.ForwardCursor(seek, opts...)
 	if err != nil {
-		return nil, 0, err
+		return nil, 0, influxdb.ErrUnexpectedTaskBucketErr(err)
 	}
 
-	var (
-		afterSeen bool
-		after     = func(task *influxdb.Task) bool {
-			if filter.After == nil || afterSeen {
-				return true
-			}
+	// free cursor resources
+	defer c.Close()
 
-			if task.ID == *filter.After {
-				afterSeen = true
-			}
+	ps, _ := s.maxPermissions(ctx, tx, *filter.User)
 
-			return false
-		}
-		matchFn = newTaskMatchFn(filter, org)
-	)
+	matchFn := newTaskMatchFn(filter, nil)
 
-	for _, m := range maps {
-		task, err := s.findTaskByIDWithAuth(ctx, tx, m.ResourceID)
-		if err != nil && err != influxdb.ErrTaskNotFound {
-			return nil, 0, err
-		}
-		if err == influxdb.ErrTaskNotFound {
-			continue
+	for k, v := c.Next(); k != nil; k, v = c.Next() {
+		kvTask := &kvTask{}
+		if err := json.Unmarshal(v, kvTask); err != nil {
+			return nil, 0, influxdb.ErrInternalTaskServiceError(err)
 		}
 
-		if matchFn == nil || matchFn(task) {
-			if !after(task) {
-				continue
+		t := kvToInfluxTask(kvTask)
+		if matchFn == nil || matchFn(t) {
+			t.Authorization = &influxdb.Authorization{
+				Status:      influxdb.Active,
+				UserID:      t.OwnerID,
+				ID:          influxdb.ID(1),
+				OrgID:       t.OrganizationID,
+				Permissions: ps,
 			}
 
-			ts = append(ts, task)
+			ts = append(ts, t)
 
 			if len(ts) >= filter.Limit {
 				break
 			}
 		}
+	}
 
+	if err := c.Err(); err != nil {
+		return nil, 0, err
 	}
```

Comment:
You'll want to check the error on `cursor.Close()` here

Reply:
I added error checking on the return.